scribejava-apis/src/main/java/com/github/scribejava/apis/KeycloakApi.java

```diff
@@ -0,0 +1,51 @@
+package com.github.scribejava.apis;
+
+import com.github.scribejava.apis.openid.OpenIdJsonTokenExtractor;
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.extractors.TokenExtractor;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+
+public class KeycloakApi extends DefaultApi20 {
+
+    private String baseUrl = "http://localhost:8080";
+    private String realm = "master";
```

Comment:
I'd prefer API classes to be immutable.
And if we need some customisation, then I'd make method like this
```
    public static FacebookApi customVersion(String version) {
        return new FacebookApi(version);
    }
```
https://github.com/scribejava/scribejava/blob/master/scribejava-apis/src/main/java/com/github/scribejava/apis/FacebookApi.java#L39

I suggest such impl from a programmatic point of view
```
public class KeycloakApi extends DefaultApi20 {

    private final String baseUrlWithRealm;

    private static final ConcurrentMap<String, KeycloakApi> INSTANCES = new ConcurrentHashMap<>();

    protected KeycloakApi(String baseUrlWithRealm) {
        this.baseUrlWithRealm = baseUrlWithRealm;
    }

    public static KeycloakApi instance() {
        return instance("http://localhost:8080/", "master");
    }

    public static KeycloakApi instance(String baseUrl, String realm) {
        final String defaultBaseUrlWithRealm = composeBaseUrlWithRealm(baseUrl, realm);

        //java8: switch to ConcurrentMap::computeIfAbsent
        KeycloakApi api = INSTANCES.get(defaultBaseUrlWithRealm);
        if (api == null) {
            api = INSTANCES.putIfAbsent(defaultBaseUrlWithRealm, new KeycloakApi(defaultBaseUrlWithRealm));
        }
        return api;
    }

    protected static String composeBaseUrlWithRealm(String baseUrl, String realm) {
        return baseUrl + (baseUrl.endsWith("/") ? "" : "/") + "auth/realms/" + realm;
    }

    @Override
    public String getAccessTokenEndpoint() {
        return baseUrlWithRealm + "/protocol/openid-connect/token";
    }

    @Override
    protected String getAuthorizationBaseUrl() {
        return baseUrlWithRealm + "/protocol/openid-connect/auth";
    }
...
}
```
Is it ok from the business side?

Reply:
and usage:
```
        final OAuth20Service service = new ServiceBuilder(apiKey)
                .apiSecret(apiSecret)
                .scope("openid")
                .callback("http://najdi.si")
                .build(KeycloakApi.instance(BASE_URL, REALM));
```